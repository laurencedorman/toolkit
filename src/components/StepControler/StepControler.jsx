import React, { Component } from 'react';

const StepController = (WrappedComponent) => {
  class Controler extends Component {
    constructor(props) {
      super(props);
      this.state = {
        steps: undefined,
        activeStep: 0,
      };
    }

    prevStep = () => {
      this.setState(prevState => ({
        activeStep: prevState.activeStep - 1,
      }));
    };

    nextStep = () => {
      this.setState(prevState => ({
        activeStep: prevState.activeStep + 1,
      }));
    };

    render() {
      const { steps, activeStep } = this.state;

      return (
        <WrappedComponent
          {...this.props}
          steps={steps}
          activeStep={activeStep}
          prevState={this.prevStep}
          nextStep={this.nextStep}
        />
      );
    }
  }
  return Controler;
};

export default StepController;
