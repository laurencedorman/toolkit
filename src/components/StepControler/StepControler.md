StepperControl with RenderChild pattern :
  
State to Props | Type | Description
-------------- | ---- | -----------
prevStep | func | Decrement (-1)
nextStep |func | Increment (+1)
setTotalStep | func | Set total children of controled target
activeStep | number | index od active step
direction | string | 'up' or 'down
lastStep | boolean | true when last step is active
totalStep | number | return total step (use setTotalStep before to set number)



```js
<StepControler>
  {({ prevStep, nextStep, activeStep, setTotalStep }) => (
    <Wrapper direction="row" style={{ alignItems: 'center', justifyContent: 'space-around' }}>
      <Button onClick={prevStep}>previous step</Button>
      <Card>{activeStep}</Card>
      <Button onClick={nextStep}>next step</Button>
    </Wrapper>
  )}
</StepControler>
```
