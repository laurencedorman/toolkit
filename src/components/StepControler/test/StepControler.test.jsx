import React from 'react';
import { mount } from 'enzyme';
import StepControler from '../StepControler';

describe('StepControler', () => {
  const renderMock = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <StepControler getActiveStep={renderMock}>
        {({ props }) => (
          <div {...props}>test</div>
        )}
      </StepControler>,
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render without crash with layoutContainer', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should set state on mount', (done) => {
    setImmediate(() => {
      expect(wrapper.state('activeStep')).toEqual(renderMock);
      done();
    }, 0);
  });
});
