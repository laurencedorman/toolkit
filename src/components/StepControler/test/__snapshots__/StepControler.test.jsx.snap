// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepControler should match snapshot 1`] = `""`;
